lib/core/src/Cardano/Wallet/Types/Read/Block.hs

```diff
@@ -0,0 +1,168 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+
+{- |
+Copyright: © 2022 IOHK
+License: Apache-2.0
+
+The 'Block' type represents blocks as they are read from the mainnet ledger.
+It is compatible with the era-specific types from @cardano-ledger@.
+-}
+module Cardano.Wallet.Types.Read.Block
+    ( Block
+    , getBlockHeight
+    , getSlotNo
+
+    , getTxsFromBlock
+    , getTxsListFromBlock
+    , numberOfTransactionsInBlock
+    ) where
+
+import Prelude
+
+import Cardano.Api
+    ( CardanoEra (..) )
+import Data.Foldable
+    ( toList )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Sequence.Strict
+    ( StrictSeq )
+import Data.Word
+    ( Word32 )
+
+import qualified Ouroboros.Consensus.Cardano.Block as O
+import qualified Ouroboros.Consensus.Protocol.Praos.Header as O
+import qualified Ouroboros.Consensus.Shelley.Ledger.Block as O
+import qualified Ouroboros.Network.Block as O
+
+import qualified Cardano.Chain.Block as Byron
+import qualified Cardano.Chain.UTxO as Byron
+import qualified Ouroboros.Consensus.Byron.Ledger.Block as Byron
+
+import qualified Cardano.Ledger.Era as Shelley
+import qualified Cardano.Ledger.Shelley.API as Shelley
+import qualified Cardano.Protocol.TPraos.BHeader as Shelley
+
+import qualified Cardano.Ledger.Alonzo.TxSeq as Alonzo
+
+import qualified Cardano.Wallet.Types.Read.Tx as Read
+
+import qualified Data.Sequence.Strict as Seq
```

Comment:
See: [Coding-Standards:Imports](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md#stylish-haskell-is-used-to-format-grouped-imports--language-pragmas)

> imports should be grouped into three groups, separated by a blank newline.
>
> * Prelude import
> * Explicit imports
> * Qualified imports

The qualified import section should be one contiguous block, without newlines:

```suggestion
import qualified Cardano.Chain.Block as Byron
import qualified Cardano.Chain.UTxO as Byron
import qualified Cardano.Ledger.Alonzo.TxSeq as Alonzo
import qualified Cardano.Ledger.Era as Shelley
import qualified Cardano.Ledger.Shelley.API as Shelley
import qualified Cardano.Protocol.TPraos.BHeader as Shelley
import qualified Cardano.Wallet.Types.Read.Tx as Read
import qualified Data.Sequence.Strict as Seq
import qualified Ouroboros.Consensus.Byron.Ledger.Block as Byron
import qualified Ouroboros.Consensus.Cardano.Block as O
import qualified Ouroboros.Consensus.Protocol.Praos.Header as O
import qualified Ouroboros.Consensus.Shelley.Ledger.Block as O
import qualified Ouroboros.Network.Block as O
```

Reply:
Then, for the qualified imports, I would like to amend our coding standards. 🤓

By arranging the qualified imports in different contiguous blocks, I was able to group them by the name of re-export (e.g. `Byron` or `O`), rather than by the original module name. I believe that this improves abstraction boundaries — I can put all Byron-related stuff behind the `Byron` qualifier in my head, and go to the `… as Byron` import block in the event that I'm interested in figuring out from which module a particular name (e.g. `Byron.foo`) was imported.

Relatedly, I would argue that the practice of putting all qualified imports together without sorting by re-export name was a large contributor to the messiness of our `Compatibility` module. For example, we had at least three re-export names for Shelley-related modules — `SL`, `SLAPI`, and `Shelley` — and it was never clear which name served which purpose. The alphabetic ordering by module name disrupted a clear grouping by re-export, and encouraged the creation of new re-export names.

In general, I would argue that namespace management is an important part of drawing and maintaining abstraction boundaries — the qualified imports section is more about semantic readability than syntactic readability.

Reply:
@HeinrichApfelmus 

Thanks for sharing your thoughts here. You've made some good points that I hadn't considered before!

Perhaps a discussion outside of GitHub might help us here?

For this PR though, perhaps we could do either of the following:
- amend the coding standards (after a discussion with the team), and then only afterwards merge this PR (assuming it follows the updated standards without modification). Perhaps there's a way to configure `stylish-haskell` so that it sorts by qualifier, instead of module path.
- amend this PR to follow the current coding standards, and then start a discussion on amending the coding standards for imports with the team (if you'd prefer to merge sooner, rather than later).

What do you think?

I would prefer for us to avoid the quick-and-easy option of just ignoring the coding standards, if possible! 😄 (Unless it's an emergency, of course!)